src/cargo/ops/cargo_rustc/job_queue.rs

```diff
@@ -245,12 +245,7 @@ impl<'a> Dependency for (&'a PackageId, Stage) {
         let (id, stage) = *self;
         let pkg = packages.iter().find(|p| p.package_id() == id).unwrap();
         let deps = resolve.deps(id).into_iter().flat_map(|a| a)
-                          .filter(|dep| *dep != id)
-                          .map(|dep| {
-                              (dep, pkg.dependencies().iter().find(|d| {
-                                  d.name() == dep.name()
-                              }).unwrap())
-                          });
```

Comment:
OK that's fine. But the cases would be rewritten like this

    base.extend(deps.filter(|&(_, dep)| dep.any(|d| d.is_build()))
                                    .map(|(id, _)| (id, Stage::Libraries)));

And the compiler compains again "cannot move out of borrowed content". I tried `ref dep` and `ref mut dep` but still it won't work out.